app/controllers/moab_storage_controller.rb

```diff
@@ -11,7 +11,13 @@ def index
   end
 
   def show
-    @output = { current_version: Stanford::StorageServices.current_version(params['id']) }
+    moab_path = path_from_object_id(params['id'])
+    object_size = directory_size(moab_path)
+    size_human = (object_size / 1_000_000.0)
```

Comment:
i would be in favor of having `object_size` be in bytes (maybe called `object_size_in_bytes`, but i think it'd be good to assume that a field called `object_size` is in bytes unless otherwise specified).

and then, we could do the human conversion in a field called `object_size_human`.  or, in this case, i might call it `object_size_in_mb`.  there is also a rails method that puts out human readable size, given a number:  https://apidock.com/rails/ActionView/Helpers/NumberHelper/number_to_human_size

which is not as consistent or easily machine-readable as always having it in MB, but i think machine-readability is what size in bytes is for anyway.

personally, i'd be slightly inclined not to include a human readable size here at all, and leave that to a display layer that we know is for humans.  although maybe that's an argument for leaving it here, since developers will probably be reading lots of REST calls while developing.

Reply:
I like the idea of having two variables object_size_in_mb and object_size_in_bytes. For now, I think i'll have the output be size in mb, but we can change that later if it is not needed. Using the ActionView Helper `number_to_human_size` converts bytes to mb however it adds "MB" after the number.

e.g 

`object_size_in_bytes = 6601408
object_size_in_mb = number_to_human_size(object_size_in_bytes)
p object_size_in_mb

>> "6.3 MB"`

Reply:
including both the `_bytes` version and the human readable version of the field sounds good to me.   

re: `number_to_human_size`:  i think adding "MB" is fine in terms of functionality (and it should presumably switch to "GB" or "KB" or whatever as appropriate), since that's just supposed to be a human readable string.  but i don't think i have an informed opinion on the performance hit of `require 'action_view'` / `include ActionView::Helpers::NumberHelper`.  my gut feeling is that it won't be substantial at the volume of requests that we're dealing with.  but i leave it up to you.  if you did go with the helper method, you'd probably want to name the field something like `object_size_human` or something, since it won't necessarily be in MB.

slight preference for the functionality of using `number_to_human_size`, but i'd also have no objection to just dividing by 1,000,000 and having the field always be `_mb` and calling it a day.

Reply:
technically, since a kilobyte is 1024, the conversion is correct.  I think it's appropriate to have the version be "correct" rather than divide by 000s.  If it weren't bytes we were talking about, I would go with our manual division.    I think using the action_view helper will be easier in the long run as I'm guessing it will cope well with KB, MB, GB, TB, etc. and I suspect our objects may vary quite a bit.

I would go with labels `object_size` and `object_size_human` but don't feel strongly about the label of the latter if someone else has a better suggestion.